File: HarvardX CS50 - Intro to Programming/craigpauga-cs50-2019-x-bleep/craigpauga-cs50-2019-x-cash/cash.c
// Craig Pauga
// 15 May 2019
// Problem Set 1

//Get the amount of change owed
#include <cs50.h>
#include <stdio.h>
#include <math.h>

//Declare cash function
void cash(void);

//Get the amount of change owed
int main(int argc, string argv[])
{
    cash();
}

void cash(void)
{
    //Initialize the change amount 
    float change;
    int n = 0;
    
    //Obtain User Input
    do
    {
        change = get_float("Change owed: ");       
    } 
    while (change <= 0);
    
    //Turn float to int
    int cash = round(change * 100);
    
    //While Loop that chips away at change until negative
    while (cash > 0)
    {   
        //Representing a quarter
        if (cash >= 25)
        {
            cash -= 25;
            n++;
        }
        //Representing a dime
        else if (cash < 25 && cash >= 10)
        {
            cash -= 10;
            n++;
        }
        //Representing a nickel
        else if (cash < 10 && cash >= 05)
        {
            cash -= 05;
            n++;
        }
        //Representing a penny
        else
        {
            cash -= 01;
            n++;
        }
    }
    //Print the answer
    printf("%i\n", n);
}

File: HarvardX CS50 - Intro to Programming/craigpauga-cs50-2019-x-sentimental-cash/cash.py
# Craig Pauga
# CS50 Cash
# Yer

from cs50 import get_float


def main():

    # first ask
    cash = get_float("Change Owed:")

    # while user does not obey master
    while cash < 0:
        cash = get_float("Change Owed:")

    # Initialize Count
    count = 0

    # multiple by 100
    cash = cash * 100

    # greedy algorithm
    while cash > 0:

        # Take a quarter
        if cash >= 25:
            cash -= 25
            count += 1

        # Take a dime
        elif (cash < 25 and cash >= 10):
            cash -= 10
            count += 1

        # Take a nickel
        elif (cash < 10 and cash >= 5):
            cash -= 5
            count += 1

        # Take a penny
        else:
            cash -= 1
            count += 1

    # Print answer
    print(count)


if __name__ == '__main__':
    main()